src/components/genre-select/genre-select.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { describe, expect, test, vi } from 'vitest'
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import userEvent from '@testing-library/user-event'
+import { GenreSelect } from './genre-select.tsx'
+
+const user = userEvent.setup()
+
+describe('Search', () => {
+  const options = ['Action', 'Comedy', 'Drama']
+  const initialSelectedOptions = 'Comedy'
+  test('Test that component renders all genres passed in props', () => {
+    const onSelect = vi.fn(() => null)
+    const { getByText } = render(
+      <GenreSelect options={options} selected={initialSelectedOptions} onSelect={onSelect} />,
+    )
```

Comment:
Mock function implementation is not needed here:

```suggestion
    const onSelect = vi.fn()
    const { getByText } = render(
      <GenreSelect options={options} selected={initialSelectedOptions} onSelect={onSelect} />,
    )
```

And you don't even need to declare the handler beforehand:

```suggestion
    const { getByText } = render(
      <GenreSelect options={options} selected={initialSelectedOptions} onSelect={vi.fn()} />,
    )
```